drivers/include/winc1500.h

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (C) 2017 Bumsik Kim <k.bumsik@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    drivers_winc1500 WINC1500
+ * @ingroup     drivers_netdev
+ * @brief       Driver for the WINC1500 WiFi module
+ * @details
+ *
+ * ## Introduction
+ * WINC1500 (or ATWINC1500) is a Wi-Fi chipset from Atmel. This chipset is
+ * designed to be used with microcontrollers(MCUs). This is offered as Wi-Fi
+ * only modules to interface with a MCU through SPI and complete MCU modules
+ * such as ATSAMW25, a heart of Arduino-mkr1000.
+ *
+ * ## Hardware/Firmware support
+ * This driver supports the following modules but not limited to:
+ *      * ATWINC1500-XPRO
+ *      * Arduino-mkr1000
+ *
+ * Currently, the supported module firmware verison is `19.5.2`. Since this
+ * firmware version is the lastest version, you may need to upgrade the firmware
+ * to use this driver. You can use [Arduino IDE (for Arduino-mkr1000)][1] or
+ * look at [Appendix D in the WINC1500 Software Design Guide][2] and
+ * [Section 5.1 in the Getting Started Guide for ATWINC1500][3] in order
+ * to update.
+ *
+ * [1]: https://www.arduino.cc/en/Tutorial/FirmwareUpdater
+ * [2]: http://www.atmel.com/Images/Atmel-42420-WINC1500-Software-Design-Guide_UserGuide.pdf
+ * [3]: http://ww1.microchip.com/downloads/en/devicedoc/atmel-42417-getting-started-guide-for-atwinc1500wifi-using-samd21-xplained-pro_userguide.pdf
+ *
+ * ## How to use the driver
+ * Firstly, you need to include `USEMODULE += winc1500` in a Makefile. This
+ * will add Wi-Fi connectivity to your board. You need to use GNRC network stack
+ * for higher level networking capability.
+ *
+ *
+ * @todo        Support for firmware version lower than 19.5.2.
+ * @todo        RIOT Socket port for the module's socket API.
+ * @todo        WPS-Enterprise support.
+ * @todo        More Wi-Fi mode support, such as Wi-Fi AP mode.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Device driver definition for the WINC1500 WiFi module.
+ *
+ * @author      Bumsik Kim <k.bumsik@gmail.com>
+ */
+#ifndef WINC1500_H
+#define WINC1500_H
+
+#include "mutex.h"
+#include "mbox.h"
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#ifdef MODULE_GNRC_NETDEV
```

Comment:
Here and below: can you check for `MODULE_GNRC`? This way it's more portable. If all goes right, `gnrc_netdev` will be deprecated in favor for `gnrc_netif2` at the end of the week.

Reply:
I see. I will test for `MODULE_GNRC`. Is `gnrc_netif2` related to #7424 ? Looks like I need to keep track of the new `gnrc_netif2`.

Reply:
> Looks like I need to keep track of the new gnrc_netif2.

If your `recv` netdev-method just outputs Ethernet frames, the `send` method expect them and all MAC is done by the device, you really don't have to (only maybe for the initialization). The abstraction of `netdev_eth` should take care of that

Reply:
Yes this Wi-Fi module will take care MAC. Good. I will focus on `netdev_eth` then.